### PR TITLE
[Deliver] Fix uses idfa limits tracking

### PR DESCRIPTION
**Pull request by minuscorp**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

This is the continuation of the problem born by the PR #10653 where it seems I introduced a bug in the issue I was trying to fix.

### Description

I've deleted the `:add_id_info_limits_tracking` from the `attr_mapping` method and added a further check the presence and value of the `:add_id_info_uses_idfa` property in the `complete!` method. When is present and its value is `true`, I manually add the required keys and values to the resulting `raw_data_clone` hash.
I've added also a test case that checks the presence of the required key and its value in the resulting `raw_data` of the submission when the `:add_id_info_uses_idfa` was set to true.

### CODE REVIEW

**Comment by mmenu-mantano** [comment]
It doesn't work unfortunately.

From what I gathered with the logs, `factory` was called before `add_id_info_uses_idfa` had a chance to be set.

You should be able to test your code without having a testing app, if you just check what is the out-going JSON. It should have a value for the key `limitsTracking`.

I edited tunes_client.rb and added `puts data.to_json` in `send_app_submission()`, I guess you can return early to avoid making the network call to iTunes Connect.

**Comment by minuscorp** [comment]
I think I'll be needing help for this... I have not enough knowledge of the whole project behavior to solve this. Either help with why `param_mapping` does not behave as expected in this case or how can I modify the built JASON before being submitted.

**Comment by minuscorp** [comment]
Got it! I think it just needs your approval @mmenu-mantano and the approval of the Fastlane team! 🚀

**Review by taijizhu — COMMENTED** [review]
Verified with fastlane 2.6.1 and xcode9.

**Comment by clementprem** [comment]
Seems to be working, any idea when this fix will be pushed to prod?

**Comment by minuscorp** [comment]
@mpirri was the reviewer of the last PR I made, maybe he can help us getting this on production as soon as possible!

**Comment by mmenu-mantano** [comment]
I can't test anymore since my app got approved, fastlane stops at an earlier stage. But it seems to be working apparently :)

File: spaceship/lib/spaceship/tunes/app_submission.rb [hunk 1]
```diff
@@ -132,6 +131,13 @@ def complete!
         end
         raw_data_clone.delete("version")
 
+        if !self.add_id_info_uses_idfa.nil? && self.add_id_info_uses_idfa == true
+          raw_data_clone.set(
```
Comment (ohayon) [inline spaceship/lib/spaceship/tunes/app_submission.rb]: Any chance you could just add a comment in here describing what this is doing? Its just a bit unclear as it is 😄

**Review by ohayon — CHANGES_REQUESTED** [review]
Hey @minuscorp thanks for the fix! Just one comment in the code and then we can merge away!

File: spaceship/lib/spaceship/tunes/app_submission.rb (hunk 1 above)
Comment (minuscorp) [reply spaceship/lib/spaceship/tunes/app_submission.rb]: There you go 😄

File: spaceship/lib/spaceship/tunes/app_submission.rb [hunk 2]
```diff
@@ -132,6 +131,15 @@ def complete!
         end
         raw_data_clone.delete("version")
 
+        # Check whether the application makes use of IDFA or not
+        # and automatically set the mandatory limitsTracking value in the request JSON accordingly.
+        if !self.add_id_info_uses_idfa.nil? && self.add_id_info_uses_idfa == true
+          raw_data_clone.set(
```
Comment (ohayon) [inline spaceship/lib/spaceship/tunes/app_submission.rb]: Thanks for the comment @minuscorp. Sorry I should have been more clear, could you put a comment above this line to explain what this method is doing?

**Review by ohayon — CHANGES_REQUESTED** [review]

File: spaceship/lib/spaceship/tunes/app_submission.rb (hunk 2 above)
Comment (minuscorp) [reply spaceship/lib/spaceship/tunes/app_submission.rb]: My bad 😅

**Review by ohayon — APPROVED** [review]

**Comment by raylillywhite** [comment]
With 2.63.0 I'm still getting error that seem related to this:
` [!] Something wrong with your Ad ID information: ["ITC.apps.idfa.errorMessage.agreeoptout", "ITC.apps.idfa.errorMessage.binaryviolatesterms"].`

Higher up was:
```[16:27:10]: Setting 'add_id_info_uses_idfa' to 'true'...
[16:27:10]: Setting 'add_id_info_serves_ads' to 'false'...
[16:27:10]: Setting 'add_id_info_tracks_install' to 'true'...
[16:27:10]: Setting 'add_id_info_tracks_action' to 'true'...
[16:27:10]: Setting 'add_id_info_limits_tracking' to 'true'...```

**Comment by minuscorp** [comment]
This is still not added in the latest release. Your issue seems to be related with the problem this PR tries to solve.

**Comment by wenjuan1982** [comment]
i occurred the same issue on xcode9.0 which fastlane 2.63.0. Any progress?

**Comment by clementprem** [comment]
@wenjuan1982 these changes are not included in 2.63.0... https://github.com/fastlane/fastlane/releases/tag/2.63.0

**Comment by minuscorp** [comment]
:+1:

**Comment by SAVeselovskiy** [comment]
It doesn't work for me or something goes wrong.
Using fastlane 2.64.1.

My submission information:
```
submission_information: {
   add_id_info_limits_tracking: true
   add_id_info_serves_ads: false
   add_id_info_tracks_action: false
   add_id_info_tracks_install: true
   add_id_info_uses_idfa: true
   content_rights_has_rights: true,
   content_rights_contains_third_party_content: false,
   export_compliance_platform: 'ios',
   export_compliance_compliance_required: false,
   export_compliance_encryption_updated: false,
   export_compliance_uses_encryption: true,
   export_compliance_is_exempt: false,
   export_compliance_contains_third_party_cryptography: false,
   export_compliance_contains_proprietary_cryptography: false,
   export_compliance_available_on_french_store: true
}
```

Logs: 
```
[12:33:52]: Setting 'add_id_info_limits_tracking' to 'true'...
[12:33:52]: Setting 'add_id_info_serves_ads' to 'false'...
[12:33:52]: Setting 'add_id_info_tracks_action' to 'false'...
[12:33:52]: Setting 'add_id_info_tracks_install' to 'true'...
[12:33:52]: Setting 'add_id_info_uses_idfa' to 'true'...
[12:33:52]: Setting 'content_rights_has_rights' to 'true'...
[12:33:52]: Setting 'content_rights_contains_third_party_content' to 'false'...
[12:33:52]: Setting 'export_compliance_platform' to 'ios'...
[12:33:52]: Setting 'export_compliance_compliance_required' to 'false'...
[12:33:52]: Setting 'export_compliance_encryption_updated' to 'false'...
[12:33:52]: Setting 'export_compliance_uses_encryption' to 'true'...
[12:33:52]: Setting 'export_compliance_is_exempt' to 'false'...
[12:33:52]: Setting 'export_compliance_contains_third_party_cryptography' to 'false'...
[12:33:52]: Setting 'export_compliance_contains_proprietary_cryptography' to 'false'...
[12:33:52]: Setting 'export_compliance_available_on_french_store' to 'true'...
```

Error:
```
Something wrong with your Ad ID information: ["ITC.apps.idfa.errorMessage.agreeoptout"]
```

Can you tell me what's wrong and what to do to find a fix?

**Comment by taquitos** [comment]
Hey @SAVeselovskiy thanks for reporting this. It would help to submit new issues instead of commenting, as this way it's easier for us to manage notifications, and have a publicly visible discussion 👍

**Comment by minuscorp** [comment]
🤔 ... I don't know how this can be failing... the tests show that the required key is actually being created in the final request as far as I can know...

**Pull request merged**